modules/dind/dind.go

```diff
@@ -0,0 +1,118 @@
+package dind
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+var (
+	// containerPorts {
+	defaultDockerDaemonPort = "2375/tcp"
+)
+
+// DinDContainer represents the Docker in Docker container type used in the module
+type DinDContainer struct {
+	testcontainers.Container
+}
+
+// Deprecated: use Run instead
+// RunContainer creates an instance of the K3s container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*DinDContainer, error) {
+	return Run(ctx, "docker:28.0.1-dind", opts...)
+}
+
+// Run creates an instance of the K3s container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DinDContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image: img,
+		ExposedPorts: []string{
+			defaultDockerDaemonPort,
+		},
+		Privileged: true,
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.CgroupnsMode = "host"
+			hc.Tmpfs = map[string]string{
+				"/run":     "",
+				"/var/run": "",
+			}
+			hc.Mounts = []mount.Mount{}
+		},
+		Cmd: []string{
+			"dockerd", "-H", "tcp://0.0.0.0:2375", "--tls=false",
+		},
+		Env: map[string]string{
+			"DOCKER_HOST": "tcp://localhost:2375",
+		},
+		WaitingFor: wait.ForLog("API listen on [::]:2375"),
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, err
+		}
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *DinDContainer
+	if container != nil {
+		c = &DinDContainer{Container: container}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+func (c *DinDContainer) Host(ctx context.Context) (string, error) {
+	return c.Container.PortEndpoint(ctx, "2375/tcp", "http")
+}
+
+// LoadImage loads an image into the DinD container.
+func (c *DinDContainer) LoadImage(ctx context.Context, image string) error {
```

Comment:
suggestion: I'd see convenient adding a `WithImages` option, which automatically loads them into the dind container using the life cycle hooks (afterStarts ?)

Reply:
I agree that it is convenient, but I won't consider it for this initial pull request.